### PR TITLE
Migrate integration tests

### PR DESCRIPTION
**Pull request by pmuens**
## What did you implement:

***Implementing Issue:*** #2377 

Move integration tests from the integration-tests-suite repository to the main Serverless repository.

## How did you implement it:

Moved the files. Furthermore the integration test we had in the Serverless repository was renamed to `simple-integration-test` and is still run (deploy-invoke-updated-invoke-remove lifecycle).

The `.travis.yml` file is also updated.

## How can we verify it:

Run `npm run simple-integration-test` and `npm run complex-integration-test`

## Todos:

- [x] Write tests
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config/commands/resources
- [x] Change ready for review message below

***Is this ready for review?:*** YES

### CODE REVIEW

**Comment by pmuens** [comment]
@eahefnawy @flomotlik just moved the integration tests over.

@eahefnawy I've not copied the SNS tests as they need some updates first.

@flomotlik I've added that the complex integration test suite should be run on every build. However it would be great if this is only done when the unit tests and the simple integration test succeeds. I've looked around in the travis docs but was unable to find a explanation on how to do this.

**Comment by flomotlik** [comment]
@pmuens Full integration test suite should imho only run on Master branch. I can configure this, but I like running the build fully once in this PR before we switch it to only run on master.

**Comment by pmuens** [comment]
Ah I see 👍  Yep that makes sense.

**Comment by eahefnawy** [comment]
@pmuens why do we need the simple integration tests? I thought the integration test suite includes a full lifecyle test in the "general" category.

**Comment by pmuens** [comment]
@eahefnawy yes, it does, but it has not included the "update" step so I just took the test we already have in place and updated it slightly.

**Comment by flomotlik** [comment]
@eahefnawy because we want to have a simple integration test running for all of our branches. This can capture stuff earlier before merging. Then we have the full suite once its merged.

File: .travis.yml [hunk 1]
```diff
@@ -22,7 +22,8 @@ install:
 script:
   - if [[ -z "$INTEGRATION_TEST" && -z "$DISABLE_TESTS" ]]; then npm test; fi
   - if [[ ! -z "$DISABLE_TESTS" && ! -z "$LINTING" && -z "$INTEGRATION_TEST" ]]; then npm run lint; fi
-  - if [[ ! -z "$INTEGRATION_TEST" && ! -z ${AWS_ACCESS_KEY_ID+x} ]]; then npm run integration-test; fi
+  - if [[ ! -z "$INTEGRATION_TEST" && ! -z ${AWS_ACCESS_KEY_ID+x} ]]; then npm run simple-integration-test; fi
+  - if [[ ! -z "$INTEGRATION_TEST" && ! -z ${AWS_ACCESS_KEY_ID+x} ]]; then npm run complex-integration-test; fi
```
Comment (flomotlik) [inline .travis.yml]: This will run the build for every branch, fixing right now

**Comment by eahefnawy** [comment]
aaah oh right understood

**Pull request merged**